### PR TITLE
Feature/tasks crud

### PR DESCRIPTION
**Pull request by Jaranag**
#8 

- Created task records to create and update.
- Created task service with crud functions.
- Created task controller with create, update and delete endpoints.
- Created a function to validate task status in our controller.

### CODE REVIEW

**Review by JiajunQiuWu — APPROVED** [review]

**Review by JoseAlfredoMiguel — APPROVED** [review]

File: src/main/java/com/grupo2/kanbanboard/controllers/TaskController.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+        }
+
+        Task taskToUpdate = optionalTask.get();
+
+        if (taskStatusIsValid(updateTaskInput.status())) {
+            Task taskUpdated = taskService.changeStatus(updateTaskInput.status(), taskToUpdate);
+            return new ResponseEntity<>(taskUpdated, HttpStatus.OK);
+        }
+        return new ResponseEntity<>("Input a valid status.", HttpStatus.NOT_FOUND);
```
Comment (SaraVictorFernandez) [inline src/main/java/com/grupo2/kanbanboard/controllers/TaskController.java:54]: Having an invalid status is not equivalent to having an invalid parameter

File: src/main/java/com/grupo2/kanbanboard/controllers/TaskController.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    @PostMapping("/tasks")
+    public ResponseEntity<?> createTasks(@RequestBody CreateTaskInput createTaskInput) {
+        if (taskStatusIsValid(createTaskInput.status())) {
+            Task taskCreated = taskService.create(createTaskInput.name(), createTaskInput.projectId(),
+                    TaskStatusEnum.valueOf(createTaskInput.status()));
+            return new ResponseEntity<>(taskCreated, HttpStatus.CREATED);
+        }
+        return new ResponseEntity<>("Input a valid status.", HttpStatus.NOT_FOUND);
```
Comment (SaraVictorFernandez) [inline src/main/java/com/grupo2/kanbanboard/controllers/TaskController.java:36]: Having an invalid status is not equivalent to having an invalid parameter

File: src/main/java/com/grupo2/kanbanboard/services/TaskService.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        this.projectRepository = projectRepository;
+
+    }
+    
+    public Task create (String TaskName, int idProject, TaskStatusEnum taskStatus) {
+        Task task = new Task();
+        task.setName(TaskName);
+        task.setStatus(taskStatus);
+        Optional<Project> optionalProject = projectRepository.findById(idProject);  
+        Project taskProject = optionalProject.get();
```
Comment (SaraVictorFernandez) [inline src/main/java/com/grupo2/kanbanboard/services/TaskService.java:30]: If the project does not exist, is it ok to throw an error?

**Review by SaraVictorFernandez — APPROVED** [review]
lgtm

**Review by SaraVictorFernandez — APPROVED** [review]

**Review by rekloyd — APPROVED** [review]

**Pull request merged**